modules/021-cni-cilium/templates/ngc-disruptive-update.yaml

```
@@ -24,25 +24,68 @@ spec:
 
     # cilium-agent image hash: {{ include "helm_lib_module_image" (list . "agentDistroless") }}
 
+    desired_sau_ds_generation={{ include "agent_daemonset_template" (list . "undefined") | sha256sum | quote }}
+
     # If there is no kubelet.conf than node is not bootstrapped and there is nothing to do
     kubeconfig="/etc/kubernetes/kubelet.conf"
     if [ ! -f "$kubeconfig" ]; then
       exit 0
     fi
 
+    # If it is the first bashible run, there is nothing to do
     if [[ "${FIRST_BASHIBLE_RUN}" == "yes" ]]; then
       exit 0
     fi
 
+    # If the migration has already succeeded, there is nothing to do
     if bb-kubectl --kubeconfig $kubeconfig get node $(bb-d8-node-name) -o json | \
-          jq -e '.metadata.annotations | has("network.deckhouse.io/cilium-1-17-migration-succeeded")' >/dev/null; then
+        jq -e '.metadata.annotations | has("network.deckhouse.io/cilium-1-17-migration-succeeded")' >/dev/null; then
+      exit 0
+    fi
+
+    # If the safe-agent-updater pod does not exist, there is nothing to do
+    if sau_pod_count="$(crictl pods --namespace d8-cni-cilium --label app=safe-agent-updater -o json | \
```

```suggestion
    if sau_pod_count="$(/opt/deckhouse/bin/crictl pods --namespace d8-cni-cilium --label app=safe-agent-updater -o json | \
```
Why crictl? Possible issues with managed k8s + our cilium? Are there any?